### ISSUE TITLE
Introspection for MySQL connector to connect to existing databases

### ISSUE THREAD

**Issue by marktani** [prisma/prisma#2506]
# Feature Request

Connecting to existing databases requires [introspection](https://www.prisma.io/docs/reference/service-configuration/data-model/introspection-aeb6diethe) to generate the according datamodel.

This feature request is used to track our progess for support of MySQL introspection.

**Comment by davesimtech** [comment]
It might be useful to output a warning/error message for the time being (e.g. connecting to existing MySQL databases is not yet implemented) before creating the introspector (in EndpointDialog.ts line 261) as at first I thought I had got the wrong mysql connection settings!

![image](https://user-images.githubusercontent.com/39625299/40546995-9e35a10e-6028-11e8-918e-bc9057f1fc18.png)

**Comment by aseem001** [comment]
Please could you share when is this feature planned for release

**Comment by chrisbull** [comment]
Something like PostGraphile (https://github.com/graphile/postgraphile) could be an interesting start. I used it to get the initial database schema structure from my clients existing database, and then exported the schemeInfo, and then only used the primary objects and rewrote them to use Prisma. It's not a 100%, but so far it's a start. Looking forward for the real prisma feature to come though!

**Comment by johndaskovsky** [comment]
Is it possible to connect to an existing MySQL database without introspection?

Any estimates for when introspection for MySQL will be ready?

**Comment by jamesmbowler** [comment]
Hey John, feel free to try out my branch, linked in my previous comment. It would be nice to have someone else test it. Just pull it (into ~/prisma), and, from your directory where you have prisma.yml, etc, run nodejs ~/prisma/cli/packages/prisma-cli/dist/index.js introspect . Let me know if you have any questions. Again, I'd love for mysql introspection to be added to prisma, asap.

**Comment by johndaskovsky** [comment]
Hey @jamesmbowler -- I'll definitely give it a try! For local development I'm running MAMP. Any suggestions for how to get the configuration working with that setup?

**Comment by jamesmbowler** [comment]
@johndaskovsky - I don't think windows should be a problem, but I don't know. This will depend on which version you are using, because the format of prisma.yml has changed, so it won't work with 1.18, I believe.

1. clone the repo to ~/prisma
2. create a directory, let's call it "testing", and cd into that directory
3. make sure you have prisma.yml and docker-compose.yml set up (see the docs for that)
4. run nodejs ~/prisma/cli/packages/prisma-cli/dist/index.js introspect

There is a really good chance this won't work, for a lot of reasons. But, I am trying to get it into the product.

**Comment by johndaskovsky** [comment]
Hey @jamesmbowler - I worked on this today. Looks like your original branch from this thread is not still available. I tested with https://github.com/jamesmbowler/prisma/tree/mysqlIntrospection_2 but wasn't able to get that working. Happy to try again once your new version is ready. Thanks!

**Comment by cr101** [comment]
A MySQL connector to connect to an existing WordPress database would be very powerful

**Comment by Lleios** [comment]
Can we know if there is progress on this feature and have an estimated release date?

**Comment by gj1118** [comment]
@marktani , Were about able to get around this problem, if so can you please let me know how ? 
Thanks

**Comment by lukasa1993** [comment]
is this still ongoing ?

**Comment by marcospgp** [comment]
Any news on the progress for this feature?

**Comment by pantharshit00** [comment]
This has been implemented in 1.32
